src/main/java/com/sekwah/advancedportals/bukkit/listeners/Listeners.java

```diff
@@ -75,12 +76,14 @@ public void onJoinEvent(PlayerJoinEvent event) {
 
     @EventHandler
     public void onWorldChangeEvent(PlayerChangedWorldEvent event) {
-        Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
+        // TODO: make relability
+        // Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
     }
 
     @EventHandler(ignoreCancelled = true)
     public void onTeleportEvent(PlayerTeleportEvent event) {
-        Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
+        // TODO: make relability
+        // Portal.cooldown.put(event.getPlayer().getName(), System.currentTimeMillis());
```

Comment:
The issue with this is the reason the global cooldown was added in the first place was to stop people who were entering the server either by bungee moving or whatever not being teleported back because they load in and are almost in the portal already.

The behavior I would prefer to see would be the original cooldown variable being altered description to explain when it is triggered and mention portal specific cooldowns. So keep the original global cooldown at least for world joins such as the top two methods.

As the code is atm you are breaking the main use of cooldowns for most people just to satisfy a few use cases that need individual cooldowns. I like the idea but just make sure you arn't completely removing existing functionality.